src/lex/cpp.rs

```diff
@@ -880,17 +894,14 @@ impl<'a> PreProcessor<'a> {
     // `#define f (a) - object macro
     fn define(&mut self, start: u32) -> Result<(), Locatable<Error>> {
         let body = |this: &mut PreProcessor| {
-            this.tokens_until_newline()
+            this.tokens_until_newline(true)
                 .into_iter()
+                .skip(1)  // First is always unwanted whitespace
```

Comment:
Are you sure this is true? What about things like `#define f+`? This has different behavior on your branch than on master (and probably deserves an automated test).

Reply:
Ah interesting, clang gives an error if there's not a space. We should do the same.

```
$ clang -x c - -E -pedantic-errors -P
#define f+
f
<stdin>:1:10: error: ISO C99 requires whitespace after the macro name [-Werror,-Wc99-extensions]
```

Reply:
"warning: ISO C99 requires whitespace after the macro name"
I guess I should at least error here if it's not whitespace instead of blindly assuming it is.

Reply:
Whitespace is no longer required between `#define` id and body. I do not give a warning if there is no whitespace, though.